exec/java-exec/src/main/java/org/apache/drill/exec/store/StoragePluginRegistryImpl.java

```diff
@@ -260,6 +289,10 @@ public StoragePlugin createOrUpdate(String name, StoragePluginConfig config, boo
 
       if (done) {
         if (persist) {
+          if (!config.isValidSecurityConfig(secureStoragePlugin)) {
+            logger.warn("Security setting for {} plugin is not recommended one. It should be set to: {}",
```

Comment:
This will cause confusion if we let the user update the storage plugin config ignoring the drill security config. I agree with Paul this can be messy, with things getting out of sync. We should discuss about this.